packages/apps/plugins/plugin-markdown/src/MarkdownPlugin.tsx

```diff
@@ -85,7 +100,12 @@ export const MarkdownPlugin = (): PluginDefinition<MarkdownPluginProvides> => {
             }
             break;
           case 'section':
-            if (datum && typeof datum === 'object' && 'object' in datum && isMarkdown(datum.object)) {
+            if (
+              datum &&
+              typeof datum === 'object' &&
+              typeof (datum as { [key: string]: any }).object === 'object' &&
```

Comment:
if you want to avoid casting, the check typescript is looking for is `'object' in datum` before allowing you to do `typeof datum.object`